### PR TITLE
Implemented: Modal to display server errors(#2gg32yc)

### PR DESCRIPTION
**Pull request by shashwatbangar**
### Related Issues
 

 Closes #

 ### Short Description and Why It's Useful
 

 ### Screenshots of Visual Changes before/after (If There Are Any)
 
[Screencast from Wednesday 31 August 2022 04:42:53  IST.webm](https://user-images.githubusercontent.com/62360781/187665957-5890ee69-a780-4cce-a55d-0da967f4d67c.webm)

 **IMPORTANT NOTICE** - Remember to add changelog entry

 ### Contribution and Currently Important Rules Acceptance
 

 - [ ] I read and followed [contribution rules](https://github.com/hotwax/threshold-management#contribution-guideline)

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -8,14 +9,41 @@ const hasError = (response: any) => {
     return !!response.data._ERROR_MESSAGE_ || !!response.data._ERROR_MESSAGE_LIST_;
 }
 
-const showToast = async (message: string) => {
+const showToast = async (message: string, err?: any) => {
+  if(message === "Something went wrong" && err) {
```
Comment (adityasharma7) [inline src/utils/index.ts:13]: `message === "Something went wrong"`  
I think this check is not needed

File: src/utils/index.ts [hunk 2]
```diff
@@ -8,14 +9,41 @@ const hasError = (response: any) => {
...
     const toast = await toastController
-        .create({
-          message,
-          duration: 3000,
-          position: 'bottom',
-        })
-      return toast.present();
+    .create({
+      message,
```
Comment (adityasharma7) [inline src/utils/index.ts:16]: I thin we could create a initialisation object 
```
{
      message,
      duration: 3000,
      position: 'bottom'
}
```

Add buttons to it if the err is passed

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/utils/index.ts [hunk 3]
```diff
@@ -35,4 +55,9 @@ const handleDateTimeInput = (dateTimeValue: any) => {
   const dateTime = DateTime.fromISO(dateTimeValue, { setZone: true}).toFormat("yyyy-MM-dd'T'HH:mm:ss")
   return DateTime.fromISO(dateTime).toMillis()
 }
-export { handleDateTimeInput, showToast, hasError, getFeature }
+
+const checkServerError = (resp: any) => {
```
Comment (adityasharma7) [inline src/utils/index.ts:59]: Improve method name

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts [hunk 4]
```diff
@@ -29,12 +29,12 @@ const actions: ActionTree<UserState, RootState> = {
           return Promise.reject(new Error(resp.data._ERROR_MESSAGE_));
         }
       } else {
-        showToast(translate('Something went wrong'));
+        showToast(translate('Something went wrong'), getResponseError(resp));
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:32]: I think this is not needed

File: src/components/SaveThresholdModal.vue [hunk 5]
```diff
@@ -232,10 +232,10 @@ export default defineComponent({
           showToast(translate('Service has been scheduled'))
           this.closeModal();
         } else {
-          showToast(translate('Something went wrong'))
+          showToast(translate('Something went wrong'), getResponseError(ifScheduleJobSuccess))
```
Comment (ymaheshwari1) [inline src/components/SaveThresholdModal.vue:235]: `ifScheduleJobSuccess` contains a boolean value and hence will not contain the server error.

File: src/store/modules/user/actions.ts [hunk 6]
```diff
@@ -29,12 +29,12 @@ const actions: ActionTree<UserState, RootState> = {
...
         console.error("error", resp.data._ERROR_MESSAGE_);
         return Promise.reject(new Error(resp.data._ERROR_MESSAGE_));
       }
     } catch (err) {
-      showToast(translate('Something went wrong'));
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:37]: I think this is not needed. We should not share any server errors to unauthenticated user

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/utils/index.ts [hunk 7]
```diff
@@ -35,4 +55,9 @@ const handleDateTimeInput = (dateTimeValue: any) => {
   const dateTime = DateTime.fromISO(dateTimeValue, { setZone: true}).toFormat("yyyy-MM-dd'T'HH:mm:ss")
   return DateTime.fromISO(dateTime).toMillis()
 }
-export { handleDateTimeInput, showToast, hasError, getFeature }
+
+const getResponseError = (resp: any) => {
+  const error = resp.error || resp.data._ERROR_MESSAGE_ || resp.data._ERROR_MESSAGE_LIST_;
```
Comment (ymaheshwari1) [inline src/utils/index.ts:60]:
```suggestion
  return resp.data.error || resp.data._ERROR_MESSAGE_ || resp.data._ERROR_MESSAGE_LIST_ || resp.data.errorMessage || resp.data.errorMessageList;
```

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

File: src/store/modules/user/actions.ts [hunk 8]
```diff
@@ -3,7 +3,7 @@ import { ActionTree } from 'vuex'
 import RootState from '@/store/RootState'
 import UserState from './UserState'
 import * as types from './mutation-types'
-import { hasError, showToast } from '@/utils'
+import { getResponseError, hasError, showToast } from '@/utils'
```
Comment (adityasharma7) [inline src/store/modules/user/actions.ts:6]: I think `getResponseError` is not used anywhere, we could remove the import

File: src/utils/index.ts [hunk 9]
```diff
@@ -35,4 +55,14 @@ const handleDateTimeInput = (dateTimeValue: any) => {
   const dateTime = DateTime.fromISO(dateTimeValue, { setZone: true}).toFormat("yyyy-MM-dd'T'HH:mm:ss")
   return DateTime.fromISO(dateTime).toMillis()
 }
-export { handleDateTimeInput, showToast, hasError, getFeature }
+
+const getResponseError = (resp: any) => {
+  if(Array.isArray(resp)) {
```
Comment (adityasharma7) [inline src/utils/index.ts:60]: I think we should handle it from where we are calling it

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**